samza-core/src/main/java/org/apache/samza/clustermanager/ContainerManager.java

```diff
@@ -233,6 +233,30 @@ void handleContainerLaunchFail(String processorId, String containerId, String pr
     }
   }
 
+  /**
+   * Handle the container stop failure for active containers and standby (if enabled).
+   * @param processorId logical id of the container eg 1,2,3
+   * @param containerId last known id of the container deployed
+   * @param containerHost host on which container is requested to be deployed
+   * @param containerAllocator allocator for requesting resources
+   */
+  void handleContainerStopFail(String processorId, String containerId, String containerHost,
+      ContainerAllocator containerAllocator) {
+    if (processorId != null && hasActiveContainerPlacementAction(processorId)) {
+      // Assuming resource acquired on destination host will be relinquished by the containerAllocator,
```

Comment:
ContainerAllocator only relinquishes resources when there are no requests in the queue. It will be worth manually releasing the resource in this case.

Reply:
There is nothing in the ContainerPlacementMetadata to determine the container-id (not processor-id) of the resources already acquired. 
The containerid in this callback is of the container whose stop failed. 

Given that there are likely many placement actions in the queue (not just one), the resource acquired may be useful for handling subsequent placement actions that are in the queue. 
Stop-fail is rare occurrence, so relying on the allocator to relinquish is perhaps not that bad.